src/Vector.cpp

```diff
@@ -0,0 +1,215 @@
+#pragma once
+#include "Vector.h"
+
+
+Vector::Vector(const size_t& size, const int& value)
+    : _size(0), _capacity(0), _data(nullptr) {
+    for (size_t i = 0; i < size; i++) {
+        push_back(value);
+    }
+}
+
+Vector::~Vector() {
+    if (_data != nullptr)
+        delete[] _data;
+}
+
+// Copy constructor
+Vector::Vector(const Vector& other)
+    : _size(0), _capacity(0), _data(nullptr) {
+    *this = other;
+}
+
+Vector& Vector::operator=(const Vector& other) {
+    if (this == &other)
+        return *this;
+
+    for (int i = 0; i < other.size(); i++) {
+        push_back(other[i]);
+    }
+    return *this;
+}
+
+// Move Constructor
+Vector::Vector(Vector&& other) noexcept
+    : _size(0), _capacity(0), _data(nullptr)
+{
+    if (this != &other) {
+        swap(other);
+    }
+}
+
+// Move Assignment Operator
+Vector& Vector::operator=(Vector&& other) noexcept
+{
+    if (this != &other) {
+        swap(other);
+    }
+    return *this;
+}
+
+// Access operator
+int& Vector::operator[](int index) {
+    if (index < 0 || index >= _size) {
+        throw std::out_of_range("Index out of bounds");
+    }
+    return _data[index];
+}
+
+const int& Vector::operator[](int index) const {
+    if (index < 0 || index >= _size) {
+        throw std::out_of_range("Index out of bounds");
+    }
+    return _data[index];
+}
+
+int* Vector::data() const {
+    return _data;
+}
+
+bool Vector::empty() const {
+    return _size == 0;
+}
+
+size_t Vector::size() const {
+    return _size;
+}
+
+void Vector::reserve(const size_t& new_cap) {
+    if (new_cap > _capacity) {
+        int* new_data = new int[new_cap];
+        for (size_t i = 0; i < _size; i++) {
+            new_data[i] = _data[i];
+        }
+
+        if (_data != nullptr)
```

Comment:
This `if` makes sense, you don't want to call `delete[]` on `nullptr`. But it isn't required since calling delete on nullptr has no effect. So the `if` can be dropped for better performance.
Check it out: https://stackoverflow.com/questions/50290022/calling-delete-or-delete-on-null-pointer